src/sentry/models/release.py

```diff
@@ -540,6 +540,19 @@ def is_valid_version(value):
     def is_semver_release(self):
         return self.package is not None
 
+    @cached_property
+    def previous_release(self):
+        """Get the release prior to this one. None if none exists"""
+        return (
+            (
+                Release.objects.filter(project_id=self.project_id, date_added__lt=self.date_added)
+                .order_by("-date_added")
+                .first()
+            )
+            if self.project_id is not None
```

Comment:
There's a note that `self.project_id` is deprecated; is that a concern?
https://github.com/getsentry/sentry/blob/c56797d1bee9dbdd21a190e8b915fdc461a04b2a/src/sentry/models/release.py#L442-L443

Reply:
Yeah actually this assumption doesn't work then. If it's a many-to-many relationship, i'll need to find a way to find the project associated with the event.

Reply:
Removed project_id usages